src/main/java/org/folio/service/inventory/InventoryUtils.java

```diff
@@ -0,0 +1,126 @@
+
+  public static final Map<String, String> INVENTORY_LOOKUP_ENDPOINTS;
+
+  private InventoryUtils() {
+  }
+
+  static {
+    INVENTORY_LOOKUP_ENDPOINTS = Map.ofEntries(
+      entry(CONTRIBUTOR_NAME_TYPES, "/contributor-name-types"),
+      entry(HOLDINGS_RECORDS, "/holdings-storage/holdings"),
+      entry(HOLDINGS_RECORDS_BY_ID_ENDPOINT, "/holdings-storage/holdings/{id}"),
+      entry(LOAN_TYPES, "/loan-types?query=name==%s&limit=1"),
+      entry(INSTANCE_STATUSES, "/instance-statuses?query=code==%s&limit=1"),
+      entry(INSTANCE_TYPES, "/instance-types?query=code==%s"),
+      entry(INSTANCES, "/inventory/instances"),
+      entry(ITEMS, "/inventory/items"),
+      entry(ITEM_BY_ID_ENDPOINT, "/inventory/items/{id}"),
+      entry(REQUESTS, "/circulation/requests"),
+      entry(HOLDINGS_SOURCES, "/holdings-sources?query=name==%s"),
+      entry(INSTANCE_RECORDS_BY_ID_ENDPOINT, "/inventory/instances/{id}")
+    );
+  }
+
+  public static Future<String> getLoanTypeId(ConfigurationEntriesCache configurationEntriesCache,
+                                             InventoryCache inventoryCache,
+                                             RequestContext requestContext) {
+    return getEntryId(configurationEntriesCache, inventoryCache, LOAN_TYPES, MISSING_LOAN_TYPE, requestContext)
+      .map(jsonObject -> jsonObject.getString(LOAN_TYPES));
+  }
+
+  public static Future<String> getSourceId(ConfigurationEntriesCache configurationEntriesCache,
+                                           InventoryCache inventoryCache,
+                                           RequestContext requestContext) {
+    return getEntryId(configurationEntriesCache, inventoryCache, HOLDINGS_SOURCES, MISSING_HOLDINGS_SOURCE_ID, requestContext)
+      .map(jsonObject -> jsonObject.getString(HOLDINGS_SOURCES));
+  }
+
+  public static Future<JsonObject> getEntryId(ConfigurationEntriesCache configurationEntriesCache,
+                                              InventoryCache inventoryCache,
+                                              String entryType,
+                                              ErrorCodes errorCode,
+                                              RequestContext requestContext) {
+    Promise<JsonObject> promise = Promise.promise();
+    getEntryTypeValue(configurationEntriesCache, entryType, requestContext)
+      .compose(entryTypeValue -> inventoryCache.getEntryId(entryType, entryTypeValue, requestContext))
+      .onSuccess(promise::complete)
+      .onFailure(t -> getEntryTypeValue(configurationEntriesCache, entryType, requestContext)
+        .onComplete(result -> {
+          if (result.succeeded()) {
+            promise.fail(new HttpException(500, buildErrorWithParameter(result.result(), errorCode)));
+
+          } else {
+            promise.fail(result.cause());
+          }
+        }));
+
+    return promise.future();
+  }
+
+  public static Future<String> getEntryTypeValue(ConfigurationEntriesCache configurationEntriesCache,
```

Comment:
This method should be kept private. Is there any other that could be changed to private after refactoring ?

Reply:
Restricted access for some more methods